sympy/solvers/solveset.py

```diff
@@ -1117,3 +1120,514 @@ def linsolve(system, *symbols):
     # Return solutions
     solution = FiniteSet(tuple(solution))
     return solution
+
+
+##############################################################################
+# ------------------------------nonlinsolve ---------------------------------#
+##############################################################################
+
+
+def substitution(system, symbols, result, known_symbols, exclude, all_symbols):
+    r"""
+    Solves the `system` using substitution method.
+
+    Parameters
+    ==========
+
+    system : list of equations.
+
+    symbols : list of unsolved symbols.
+
+    known_symbols : list of solved symbols.
+
+    result : If non of the symbols are solved then it is empty list, otherwise
+    list of already solved_syms symbols( dict symbol : value )
+
+    exclude : Set of expression which should not be zero with result.
+    Mostly denominator expression.
+
+    all_symbols : known_symbols + unsolved symbols.
+
+    Returns
+    =======
+    Returns a FiniteSet of ordered tuple.
+
+    Examples
+    ========
+
+    >>> from sympy.core.symbol import symbols
+    >>> x, y = symbols('x, y', real = True)
```

Comment:
`real = True` --> `real=True` Please follow pep8.
Please fix pep8 Errors.
Tip: Use a pep8 plugin to check for pep8 errors for your code editor.